controllers/core/openstackcontrolplane_controller.go

```diff
@@ -164,6 +166,11 @@ func (r *OpenStackControlPlaneReconciler) Reconcile(ctx context.Context, req ctr
 			condition.UnknownCondition(corev1beta1.OpenStackControlPlaneNovaReadyCondition, condition.InitReason, corev1beta1.OpenStackControlPlaneNovaReadyInitMessage),
 		)
 
+		// Initialize optional service conditions if enabled
+		if instance.Spec.Ironic.Enabled {
```

Comment:
+1 on this. It will ensure the overall deployment doesn't get stuck reporting Ironic not ready if the service is disabled. Thanks for updating the implementation here.

Reply:
Chur